src/plugins/shared-variables/shared-variables.ts

```diff
@@ -49,6 +68,12 @@ export const SharedVariables = SharedModel.named("SharedVariables")
     self.addVariable(variable);
     return variable;
   },
+  setupCompute(xName: string, yName: string) {
+    self.variablesCopy = types.array(Variable).create(getSnapshot(self.variables));
```

Comment:
I was thinking the shared variables model would need to support multiple overlapping calls to setupCompute. So for example if there are multiple traces being shown on the same graph, they might call setup setupCompute and then later call computeY asynchronously. 

To support this, the variablesCopy, computeY function, and disposeCompute function shouldn't be stored on self. Instead they'd just be scoped (local variables) to this setupCompute function.  

That change might impact performance. I'm not sure if it will be negative, positive, or nothing. So I'd recommend not changing this in this PR. But it could be tried during the work on the "add series" button.